index.ts

```diff
@@ -1,15 +1,84 @@
 import { RGBot } from "rg-bot";
+import RGCTFUtils, { CTFEvent } from "rg-ctf-utils";
+import { Vec3 } from "vec3";
 
 /**
- * This strategy is the simplest example of how to get started with the rg-bot package.
+ * This strategy is the simplest example of how to get started 
+ * with the rg-bot and rg-ctf-utils packages. The Bot will get 
+ * the flag and then run back to base to score.
+ * 
+ * Ways to extend this code:
+ * TODO: What happens when a bot is completing an action and 
+ *       another event happens?
+ * TODO: How do we respond to item spawn and drop events?
+ * TODO: How do we target and attack enemies?
+ * TODO: What different states is my bot in, and how can I organize
+ *       its behavior based on these states?
+ * rg-bot docs: https://github.com/Regression-Games/RegressionBot/blob/main/docs/api.md
+ * rg-ctf-utils docs: https://github.com/Regression-Games/rg-ctf-utils
  */
 export function configureBot(bot: RGBot) {
 
-    bot.setDebug(true);
+  // Since most blocks can't be broken on Arctic Algorithm Field,
+  // don't allow digging while pathfinding
+  bot.allowDigWhilePathing(false);
 
-    // The bot says a message when spawning
-    bot.on('spawn', async () => {
-        bot.chat('Hello, I have arrived!');
-    });
+  // Instantiate our helper utilities and events for Capture the Flag
+  const rgctfUtils = new RGCTFUtils(bot);
+
+  // track how many times we've died
+  let deaths = 0
+  
+  bot.on('death', () => {
+    console.log("I have died...")
+    ++deaths;
+    try {
+      // stop any current pathfinding goal
+      // @ts-ignore
+      bot.mineflayer().pathfinder.setGoal(null)
+      // @ts-ignore
+      bot.mineflayer().pathfinder.stop()
+    } catch (ex) {
+  
+    }
+  })
+
+  bot.on('spawn', async () => {
+    await rgctfUtils.approachFlag();
+  });
+
+  // When a player obtains the flag, this event gets called.
+  // In the case where that player is this bot, the bot
+  // navigates back to their scoring location.
+  bot.on(CTFEvent.FLAG_OBTAINED, async (collector: string) => {
+    if (collector == bot.username()) {
+      await rgctfUtils.scoreFlag()
+    }
+  });
+
+  let isCollectingItems = false;
+
+  // If the flag was scored, collect items until the flag is available
+  bot.on(CTFEvent.FLAG_SCORED, async (teamName: string) => {
+    let previousDeaths = deaths
+	  const codeStillRunning = () => {return previousDeaths === deaths}
+    bot.chat(`Flag scored by ${teamName} team, collecting items until new flag is here`)
+    isCollectingItems = true;
+    while(rgctfUtils.getFlagLocation() === null && codeStillRunning()) {
+      await bot.findAndCollectItemsOnGround();
+      await bot.waitForMilliseconds(500);
```

Comment:
This logic is iffy.  If a user picks up the flag in this 500 ms, you'll miss it and just keep collecting.  Even if they drop it again, unless you're inside 30 blocks, you'll never know.  Similarly .. when flag location becomes non null.. the flag_available event already happened, so I think your bot is just gonna stand still until the next flag scored

Reply:
a few things
1. Is the case where you just keep collecting because in the 500 ms you missed the flag spawn before someone grabbed it... this is 'fine', not ideal, but at least the bot is still doing work
2. In the case where you do see the flag drop, but a user picks it up on before you call approachFlag, your bot will just stand still. (this is a small timing window, but definitely possible based on my experiences testing bots)
3. In general, our samples should be airtight in their functionality.  We are going to have so many questions to answer that quirks in our examples shouldn't be one of them

Reply:
Understood - will be addressing and updating these ASAP